scapy/contrib/gxrp.py

```diff
@@ -0,0 +1,228 @@
+
+        normative references:
+            - IEEE 802.1D 2004 - Media Access Control (MAC) Bridges
+            - IEEE 802.1Q 1998 - Virtual Bridged Local Area Networks
+
+"""
+from scapy.fields import (
+    LenField,
+    EnumField,
+    ByteField,
+    PacketListField,
+    ShortField,
+    MACField,
+)
+from scapy.packet import Packet, bind_layers, split_layers
+from scapy.layers.l2 import LLC, Dot3
+from scapy.error import warning
+
+
+class GVRP(Packet):
+    """
+    GVRP
+    """
+
+    name = "GVRP"
+
+    # IEEE802.1Q-1998 11.2.3.1.3
+    fields_desc = [ShortField("vlan", 1)]
+
+    def extract_padding(self, s):
+        return b"", s
+
+
+class GMRP_GROUP(Packet):
+    """
+    GMRP Group
+    """
+
+    name = "GMRP Group"
+
+    # IEEE802.1D-2004 10.3.1.4
+    fields_desc = [MACField("addr", None)]
+
+    def extract_padding(self, s):
+        return b"", s
+
+
+class GMRP_SERVICE(Packet):
+    """
+    GMRP Service
+    """
+
+    name = "GMRP Service"
+
+    # IEEE802.1D-2004 10.3.1.4
+    fields_desc = [
+        EnumField(
+            "event",
+            0,
+            {0x0: "All Groups", 0x1: "All Unregistered Groups"},
+            fmt="B",
+        )
+    ]
+
+    def extract_padding(self, s):
+        return b"", s
+
+
+class GARP_ATTRIBUTE(Packet):
+    """
+    GARP attribute container
+    """
+
+    name = "GARP Attribute"
+
+    # IEEE802.1D-2004 12.10.2.4-5
+    fields_desc = [
+        LenField("len", None, fmt="B", adjust=lambda l: l + 2),
+        EnumField(
+            "event",
+            0,
+            {
+                0x0: "LeaveAll",
+                0x1: "JoinEmpty",
+                0x2: "JoinIn",
+                0x3: "LeaveEmpty",
+                0x4: "LeaveIn",
+                0x5: "Empty",
+            },
+            fmt="B",
+        ),
+    ]
+
+    def do_dissect(self, s):
+        s = super(GARP_ATTRIBUTE, self).do_dissect(s)
+        if self.len is not None and self.event == 0 and self.len > 2:
+            warning("Non-empty payload at LeaveAll event")
+        return s
+
+    def extract_padding(self, s):
+        boundary = self.len - 2
+        return s[:boundary], s[boundary:]
+
+    def guess_payload_class(self, payload):
+        try:
+            garp_message = self.parent
+            garp = garp_message.parent
+            llc = garp.underlayer
+            dot3 = llc.underlayer
+            if (
+                dot3.dst == "01:80:c2:00:00:21"
+            ):  # IEEE802.1D-2004 12.4 Table 12-1
+                return GVRP
+            elif (
+                dot3.dst == "01:80:c2:00:00:20"
+            ):  # IEEE802.1D-2004 12.4 Table 12-1
+                if garp_message.type == 1:  # IEEE802.1D-2004 10.3.1.3
+                    return GMRP_GROUP
+                elif garp_message.type == 2:  # IEEE802.1D-2004 10.3.1.3
+                    return GMRP_SERVICE
+        except AttributeError:
+            pass
+        Packet.guess_payload_class(self, payload)
```

Comment:
```suggestion
        return super(GARP_ATTRIBUTE, self).guess_payload_class(payload)
```

Missing the return + might want to use super

Reply:
done